### ISSUE TITLE
Install error

### ISSUE THREAD

**Issue by tarelli** [openworm/org.geppetto.python#3]
Hi @aranega I'm finally having the opportunity to use this in a project. I just tried installing on my system and I'm getting the following, any idea what the problem might be? 

Also we have some libraries in the project that apparently are supporting only Python 2.7 and while I'm trying to see if it would be feasible for them to support Python 3.3 I'm wondering also what would be the chances of pyecore being backward compatible to Python 2.7?
Thanks!

```
 matteocantarelli@Matteos-MBP-7  ~/Documents/Development/pygeppetto   master v0.3.5-alpha  python3 setup.py install
running install
running bdist_egg
running egg_info
writing dependency_links to pygeppetto.egg-info/dependency_links.txt
writing pygeppetto.egg-info/PKG-INFO
writing requirements to pygeppetto.egg-info/requires.txt
writing top-level names to pygeppetto.egg-info/top_level.txt
reading manifest file 'pygeppetto.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
writing manifest file 'pygeppetto.egg-info/SOURCES.txt'
installing library code to build/bdist.macosx-10.6-intel/egg
running install_lib
running build_py
creating build
creating build/lib
creating build/lib/model
copying model/__init__.py -> build/lib/model
copying model/model.py -> build/lib/model
creating build/lib/model/datasources
copying model/datasources/__init__.py -> build/lib/model/datasources
copying model/datasources/datasources.py -> build/lib/model/datasources
creating build/lib/model/types
copying model/types/__init__.py -> build/lib/model/types
copying model/types/types.py -> build/lib/model/types
creating build/lib/model/values
copying model/values/__init__.py -> build/lib/model/values
copying model/values/values.py -> build/lib/model/values
creating build/lib/model/variables
copying model/variables/__init__.py -> build/lib/model/variables
copying model/variables/variables.py -> build/lib/model/variables
creating build/bdist.macosx-10.6-intel
creating build/bdist.macosx-10.6-intel/egg
creating build/bdist.macosx-10.6-intel/egg/model
copying build/lib/model/__init__.py -> build/bdist.macosx-10.6-intel/egg/model
creating build/bdist.macosx-10.6-intel/egg/model/datasources
copying build/lib/model/datasources/__init__.py -> build/bdist.macosx-10.6-intel/egg/model/datasources
copying build/lib/model/datasources/datasources.py -> build/bdist.macosx-10.6-intel/egg/model/datasources
copying build/lib/model/model.py -> build/bdist.macosx-10.6-intel/egg/model
creating build/bdist.macosx-10.6-intel/egg/model/types
copying build/lib/model/types/__init__.py -> build/bdist.macosx-10.6-intel/egg/model/types
copying build/lib/model/types/types.py -> build/bdist.macosx-10.6-intel/egg/model/types
creating build/bdist.macosx-10.6-intel/egg/model/values
copying build/lib/model/values/__init__.py -> build/bdist.macosx-10.6-intel/egg/model/values
copying build/lib/model/values/values.py -> build/bdist.macosx-10.6-intel/egg/model/values
creating build/bdist.macosx-10.6-intel/egg/model/variables
copying build/lib/model/variables/__init__.py -> build/bdist.macosx-10.6-intel/egg/model/variables
copying build/lib/model/variables/variables.py -> build/bdist.macosx-10.6-intel/egg/model/variables
byte-compiling build/bdist.macosx-10.6-intel/egg/model/__init__.py to __init__.cpython-35.pyc
byte-compiling build/bdist.macosx-10.6-intel/egg/model/datasources/__init__.py to __init__.cpython-35.pyc
byte-compiling build/bdist.macosx-10.6-intel/egg/model/datasources/datasources.py to datasources.cpython-35.pyc
byte-compiling build/bdist.macosx-10.6-intel/egg/model/model.py to model.cpython-35.pyc
byte-compiling build/bdist.macosx-10.6-intel/egg/model/types/__init__.py to __init__.cpython-35.pyc
byte-compiling build/bdist.macosx-10.6-intel/egg/model/types/types.py to types.cpython-35.pyc
byte-compiling build/bdist.macosx-10.6-intel/egg/model/values/__init__.py to __init__.cpython-35.pyc
byte-compiling build/bdist.macosx-10.6-intel/egg/model/values/values.py to values.cpython-35.pyc
byte-compiling build/bdist.macosx-10.6-intel/egg/model/variables/__init__.py to __init__.cpython-35.pyc
byte-compiling build/bdist.macosx-10.6-intel/egg/model/variables/variables.py to variables.cpython-35.pyc
creating build/bdist.macosx-10.6-intel/egg/EGG-INFO
copying pygeppetto.egg-info/PKG-INFO -> build/bdist.macosx-10.6-intel/egg/EGG-INFO
copying pygeppetto.egg-info/SOURCES.txt -> build/bdist.macosx-10.6-intel/egg/EGG-INFO
copying pygeppetto.egg-info/dependency_links.txt -> build/bdist.macosx-10.6-intel/egg/EGG-INFO
copying pygeppetto.egg-info/requires.txt -> build/bdist.macosx-10.6-intel/egg/EGG-INFO
copying pygeppetto.egg-info/top_level.txt -> build/bdist.macosx-10.6-intel/egg/EGG-INFO
zip_safe flag not set; analyzing archive contents...
creating dist
creating 'dist/pygeppetto-0.1.0-py3.5.egg' and adding 'build/bdist.macosx-10.6-intel/egg' to it
removing 'build/bdist.macosx-10.6-intel/egg' (and everything under it)
Processing pygeppetto-0.1.0-py3.5.egg
Copying pygeppetto-0.1.0-py3.5.egg to /Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages
Adding pygeppetto 0.1.0 to easy-install.pth file

Installed /Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/pygeppetto-0.1.0-py3.5.egg
Processing dependencies for pygeppetto==0.1.0
Searching for pyecore>=0.2.0
Reading https://pypi.python.org/simple/pyecore/
Best match: pyecore 0.7.1
Downloading https://pypi.python.org/packages/8d/22/a5ed73e367e448124a2981556ec18e355c155268e47350e879cbd65a9d93/pyecore-0.7.1.tar.gz#md5=09f4376bbac1c9b592a02ddecd1cc732
Processing pyecore-0.7.1.tar.gz
Writing /var/folders/dr/p3gpq2q91nj7skf5f5qw0fd00000gn/T/easy_install-n__dj1m9/pyecore-0.7.1/setup.cfg
Running pyecore-0.7.1/setup.py -q bdist_egg --dist-dir /var/folders/dr/p3gpq2q91nj7skf5f5qw0fd00000gn/T/easy_install-n__dj1m9/pyecore-0.7.1/egg-dist-tmp-a1pjadwv
error: Setup script exited with error in pyecore setup command: 'install_requires' must be a string or list of strings containing valid project/version requirement specifiers; Expected version spec in enum34;python_version<"3.4" at ;python_version<"3.4"
```

**Comment by aranega** [comment]
Hi @tarelli, I performed many PyEcore upgrade lately, each time I check if pygeppetto tests are ok, but I didn't check if the install work. It seems that the issue would come from `setuptools` which does not understand one of the dependency format. I already had this issue on an outdated version of `setuptool`, while I try to replicate the issue, could you try to perform a `setuptools` upgrade?

```bash
$ pip install setuptools -U
```
I can check if I can make PyEcore Python 2.7 friendly, but from what I recal, there where issues with some 'on the fly' modification that are performed on in-memory classes. I will check anyway, the code moved a lot, so you never know.

_[not directly related]_
In the meantime, I'm also working on two points that could be interesting for you:

1. a command line script for you to generate more easily the pygeppetto code from the `.ecore` without using Acceleo (using [pyecoregen](https://github.com/pyecore/pyecoregen), developed by @moltob). I started to write it few weeks ago, it is almost finished, but there is some corner cases I need to tackle.

2. the integration of PyEcore in [textX](https://github.com/igordejanovic/textX), a parser generator written in Python. If I remember correctly, you were interested in providing a textual syntax for pygeppetto, this could help you to do it in Python.

_[/not directly related]_

**Comment by tarelli** [comment]
@aranega I already tried updating setuptools (it was the first thing I found on SO but even with the latest `Successfully installed setuptools-36.6.0` I still have the same error), I'll keep looking into it.
In the meanwhile I learnt we need to have compatibility with python 2.7 unfortunately so it would be very good if PyEcore could be backward compatible or I don't know if I'd be able to use it for now :(
Thanks also for the other updates, indeed good to know, I'll keep both in mind!

**Comment by aranega** [comment]
Arf, I was secretly hoping that the upgrade would do the trick. Currently, I didn't find a way to replicate the bug, which Python 3 version are you using? I will build a virtualenv with the same version than yours. 

Also, which Python version is your system default one? If the default one is Python 2, did you use `pip3` to perform `setuptools` upgrade? (I'm not sure this is relevant, but if pip is associated to Python 2, perhaps the upgrade was fine for Python 2, but not for Python 3?)

I will check really quickly the effort required to make PyEcore Python 2.7 compatible, I hope it will work!

**Comment by tarelli** [comment]
Default is Python 2 but I did upgrade both pip and pip3. I have Python 3.5.0.

**Comment by aranega** [comment]
No luck, I cannot replicate your issue, I tried with Python 2 as default version and using pip3 to install pyecore and pygeppetto, it worked. I also built a Python 3.5 virtualenv (I'm using 3.6), but still, everything installed just fine. Could it be your pip version? What gives you `pip3 --version`? Perhaps I can try to match your exact pip version also.

I'm trying to see if PyEcore could be ported, I'm not certain, but I think so. The generated code should be modified, and the modifications I'm doing are not very elegant at the moment.

**Comment by tarelli** [comment]
thanks @aranega I'll keep investigating that on my side no worries, I'll be happy enough if everything becomes 2.7 compatible! :)

**Comment by aranega** [comment]
Good news! I finally made PyEcore working with Python 2.7. Modifications were less difficult than what I tought as the code evolved a lot from the last time I tried. I had to perform many modifications though. I put them on a separate branch for now (I ported some modifications to the Python 3 version). You can check the branch [`feature/python2.7`](https://github.com/pyecore/pyecore/tree/feature/python2.7). 

I prefer to keep the 2.7 version separated from the 3.x one, so I will fork PyEcore and create a different package on pypi named `pyecore-py2`. It will be easier for me to let this version live, and continue to work on the version 3.x when Python 2.7 support will end. I also will backports new features/fix from the Python 3.x version to the 2.7.

However, it still implies a modification of the code generated for `pygeppetto`. I will try to add a switch in `pyecoregen` (which will still relies on Python 3) to generate Python 2.7 compatible code if required.

Stay tuned, I will try to do this asap.

**Comment by tarelli** [comment]
@aranega that's great Vincent, thanks so much! Let me know when it's ready for me to test!

**Comment by aranega** [comment]
@tarelli I had little time this morning so I created a dedicated repository for [PyEcore 2.7 backport](https://github.com/pyecore/pyecore-py2), and released a version of `pyecore-py2` on pypi (I changed the pypi package name to avoid confusion between the Python 3 and 2 version). I also modified the pygeppetto generated code on the `developement` branch in order to use PyEcore 2.7. Tests are running fine, but that's possible that I missed some points.

Currently, the Acceleo generator have been also changed so it generates Python 2 compatible code for PyEcore, but the prefered cli generator: `pyecoregen` have not been updated for Python 2 compatible code generation (yet).

Tell me if installation works better this time and if everything runs fine for you.
